### PR TITLE
covers issue #19124 Migrate components to TS: PickerNetwork 

### LINKED ISSUE #19124: Migrate components to TS: PickerNetwork
### Description
Migrate `PickerNetwork` component  in `ui/components/component-library` to TypeScript

Dependencies
- https://github.com/MetaMask/metamask-extension/issues/19239

### Technical Details

- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` (name and keys)
- Ensure enums use singular not plural e.g. `Size` not `Sizes`
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation

### Acceptance Criteria

- Component has been converted to TypeScript
- Stories, documentation and tests have been updated inside of the `component-name` folder
- Before/After screen cast of component stories in storybook to show no visual regression after migration

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with TypeScript and want to contribute to improving the stability of the extension

### PR DESCRIPTION
**Pull request by kamalika0363**
## Explanation
* Changed the files in ui/components/component-library to typescript
* Re-wrote everything in Typescript and also added proper extensions along with documentation.
*  Fixes #19124 

## Pre-merge author checklist

- I've clearly explained:
  - [ ✅] What problem this PR is solving
  - [ ✅] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by doesnots** [comment]
> **CLA Signature Action:**
> 
> Thank you for your submission, we really appreciate it. We ask that you read and sign our [Contributor License Agreement](https://metamask.io/cla.html) before we can accept your contribution. You can sign the CLA by just by adding a comment to this pull request with this exact sentence:
> 
> > _**I have read the CLA Document and I hereby sign the CLA**_
> 
> By commenting with the above message you are agreeing to the terms of the CLA. Your account will be recorded as agreeing to our CLA so you don't need to sign it again for future contributions to this repository.
> 
> **0** out of **1** committers have signed the CLA. ❌ @doesnots

I have read the CLA Document and I hereby sign the CLA

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @doesnots, thanks for your contribution. The issue this PR is intended to solve is to replace the picker-network component this PR is creating a new folder see screenshot. You are also missing some files outlined in the template in the issue. e.g. `picker-network.types.js`

Screenshot 2023-05-16 at 10 43 04 AM

**Comment by doesnots** [comment]
My apologies, I'll correct all of that by today.
Thanks for pointing out.

File: ui/components/component-library/picker-network/README.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';
```
Comment (georgewrmarshall) [inline ui/components/component-library/picker-network/README.tsx:1]: This should remain a `.mdx` file

File: ui/components/component-library/picker-network/picker-network.tsx [hunk 2]
```diff
@@ -57,29 +57,4 @@ export const PickerNetwork = ({
...
-  /**
-   * The text content of the PickerNetwork component
-   */
-  label: PropTypes.string.isRequired,
-  /**
-   * PickerNetwork accepts all the props from Box
-   */
-  ...Box.propTypes,
-};
+export default PickerNetwork;
```
Comment (georgewrmarshall) [inline ui/components/component-library/picker-network/picker-network.tsx:60]: Should be a named export

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @doesnots, I've made a few suggestions but this PR has a fair way to go to meet the acceptance criteria.
- The component should be a named export
- The propType comments should be moved to the types in the `.types` file
- The storybook file is missing storybook types


It is also failing a lot of unit tests. I'm afraid I am going to have to close it. I appreciate your contribution and encourage you to contribute in future. These issues may be more suitable to work on.

- https://github.com/MetaMask/metamask-extension/issues/18896
- https://github.com/MetaMask/metamask-extension/issues/18714
- https://github.com/MetaMask/metamask-extension/issues/17670

**Pull request closed without merging**